BUILD

```diff
@@ -1301,3 +1301,14 @@ exports_files([
     "conformance/text_format_failure_list_java.txt",
     "conformance/text_format_failure_list_java_lite.txt",
 ])
+
+filegroup(
+    name = "bzl_srcs",
+    srcs = glob(["**/*.bzl"]),
```

Comment:
One thing that might be misleading about the globs here is that they do not actually include all `*.bzl` files in the project. This would only include all `*.bzl` files within the current [package](https://docs.bazel.build/versions/main/build-ref.html#packages).

Again, this change just means I can use the protobuf macros in files I'm generating documentation for

Reply:
**/* is a recursive glob, it's everything from that definition point and on that ends in .bzl at any tree level.

Reply:
No, [glob](https://docs.bazel.build/versions/main/be/functions.html#glob) does not traverse package boundaries. And all `.bzl` files are accompanied by a `BUILD` file
```
protobuf % find . -name "BUILD" -or -name "*.bzl"
./cc_proto_blacklist_test.bzl
./util/python/BUILD
./protobuf_version.bzl
./internal.bzl
./java/core/BUILD
./java/util/BUILD
./java/lite/BUILD
./java/internal/testing.bzl
./java/internal/BUILD
./java/BUILD
./protobuf.bzl
./third_party/BUILD
./protobuf_deps.bzl
./examples/BUILD
./benchmarks/datasets/google_message1/proto2/BUILD
./benchmarks/datasets/google_message1/proto3/BUILD
./benchmarks/datasets/google_message4/BUILD
./benchmarks/datasets/google_message3/BUILD
./benchmarks/datasets/google_message2/BUILD
./benchmarks/datasets/BUILD
./benchmarks/cpp/BUILD
./benchmarks/BUILD
./compiler_config_setting.bzl
./BUILD
./objectivec/BUILD
```

Reply:
Right, I should say, our BUILD/package boundaries are not consistent/what we want them to be long term, so I'm afraid that will be overly eager and allow non-stardoc dependencies on these files as file inputs to other targets potentially without realizing/being careful to exclude ones we don't ever intend to make accessible.

Reply:
The reason these globs are broad is to hopefully allow them to be invisible and conform to whatever structure you end up with.

Concretely, the problem I'm facing is that a project I work on introduced a new `proto` rule which loads [protobuf.bzl](https://github.com/protocolbuffers/protobuf/blob/79370f1ffa81a8394c41a99a32ab1452f923242e/protobuf.bzl) for the implementation and [protobuf_deps.bzl](https://github.com/protocolbuffers/protobuf/blob/79370f1ffa81a8394c41a99a32ab1452f923242e/protobuf_deps.bzl) to expose dependencies. This change broke our ability to generate documentation because `stardoc` requires access to these files as targets. Unless [stardoc](https://github.com/bazelbuild/stardoc) is refactored, any user of it will require all `.bzl` files they load to be accessible as Bazel targets to pass to the `stardoc` rule. Many rules expose similar targets ([rules_nodejs](https://github.com/bazelbuild/rules_nodejs/blob/stable/BUILD.bazel#L31-L59), [rules_python](https://github.com/bazelbuild/rules_python/blob/main/BUILD#L18-L52), [rules_rust](https://github.com/bazelbuild/rules_rust/blob/main/rust/BUILD.bazel#L5-L11), [rules_go](https://github.com/bazelbuild/rules_go/blob/master/go/BUILD.bazel#L16-L24)). The thought in keeping this pattern broad was that the structure hasn't changed in some time and and if it should, the pattern would account for that.